### PR TITLE
PCHR-2490: Document modal v2 update

### PR DESCRIPTION
**Pull request by bmnepali**
## Overview
After the document updates were as per v2 requirements. When admin creates a document with no files attached the status changes to "Approved" where the status should be "Awaiting Upload". This PR fixes this issue.d

## Before
- The  document status is hidden on first in the list.
before-2490-min

- Document status is changed to approved after the page reload. 
screen shot 2017-09-13 at 5 40 10 pm

## After
- Document status is shown after creating the document and status is changed to "awaiting upload"
2490-after-min

## Technical Details
1. Determine document status based on if the attachments are available.
```js
function getDocumentStatus () {
      return vm.isRole('admin')
        ? (vm.files.length || vm.uploader.queue.length ? vm.getStatusIdByName('approved') : vm.getStatusIdByName('awaiting upload'))
        : (vm.files.length || vm.uploader.queue.length ? vm.getStatusIdByName('awaiting approval') : vm.getStatusIdByName('awaiting upload'));
}
```
2. Separate the code to get the status id by name
```js
function getStatusIdByName (statusName) {
      return _.findKey($rootScope.cache.documentStatus.obj, function (status) {
        return status === statusName;
      });
}
```
## Comments
- Related test cases have been fixed as required.
- This PR is target to epic branch "PCHR-2112-onboarding-and-exiting-workflow"  instead of "staging". Once all the issue in epic branch are passed testing we merge it to staging.

- [x] Tests Pass

### CODE REVIEW

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js [hunk 1]
```diff
@@ -200,7 +200,9 @@ define([
 
         $scope.$broadcast('ct-spinner-show');
 
-        documentStatus = vm.isRole('admin') ? '3' : (vm.files.length || vm.uploader.queue.length ? '2' : '1');
+        documentStatus = vm.isRole('admin') ?
+          (vm.files.length || vm.uploader.queue.length ? '3' : '1') :
```
Comment (deb1990) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js]: Please dont use static numbers like 3 or 1, its impossible to know what they are. You must be getting all the status's from backend. So you can create a function called `getStatusIdByName()`, this function searches for the ID given the name, so you can use `(vm.files.length || vm.uploader.queue.length ?  getStatusIdByName('<>'): getStatusIdByName('<>')) :`

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js [hunk 2]
```diff
@@ -200,7 +200,9 @@ define([
 
         $scope.$broadcast('ct-spinner-show');
 
-        documentStatus = vm.isRole('admin') ? '3' : (vm.files.length || vm.uploader.queue.length ? '2' : '1');
+        documentStatus = vm.isRole('admin') ?
```
Comment (deb1990) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js]: A function can be created to abstract this, `initDocumentStat()` which will abstract this logic. That will make it more readable.

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js [hunk 3]
```diff
@@ -293,6 +290,33 @@ define([
...
+     */
+    function getDocumentStatus () {
+      return vm.isRole('admin')
+        ? (vm.files.length || vm.uploader.queue.length ? vm.getStatusIdByName('approved') : vm.getStatusIdByName('awaiting upload'))
+        : (vm.files.length || vm.uploader.queue.length ? vm.getStatusIdByName('awaiting approval') : vm.getStatusIdByName('awaiting upload'));
+    }
+
+    /**
+     * Gets the status id of a documents status based on it name
+     * @param  {string} statusName
```
Comment (deb1990) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js]: Same as above

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js [hunk 4]
```diff
@@ -293,6 +290,33 @@ define([
...
+     * @return {string|int}
+     */
+    function getStatusIdByName (statusName) {
+      var statusId = '';
+
+      _.find($rootScope.cache.documentStatus.obj, function (status, id) {
```
Comment (deb1990) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js]: This is not how `find` is supposed to work. It should be something like this
```
var status = _.find($rootScope.cache.documentStatus.obj, function (status) {
	return status.name === statusName;
});

return status ? status.id : false;
```

Comment (bmnepali) [reply uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js]: The following works:
```js
function getStatusIdByName (statusName) {
      return _.findKey($rootScope.cache.documentStatus.obj, function(status) {
        return status === statusName;
      });
    }
```

**Pull request merged**